scripts/import/dbSNP_v2/update_new_sets.pl

```diff
@@ -295,6 +312,61 @@ sub dump_new_variation_feature {
   
 }
 
+sub get_structure {
+  my $dbhvar = shift;
+
+  my $get_struc_sth = $dbhvar->prepare(qq[ select variation_set_sub, variation_set_super from variation_set_structure]);
+
+  my %parent;
+  $get_struc_sth->execute() ||die;
+  my $dat = $get_struc_sth->fetchall_arrayref();
+  foreach my $l(@{$dat}){
+    $parent{$l->[0]} = $l->[1] ;
+  }
+  return \%parent;
+
+}
+
+sub recalculate {
+  my $input_file = shift;
+  my $output_file = shift;
+  
+  my $parent = get_structure($dbh);
+  my %concat_sets;
+  
+  open FH, "<", "$input_file" or die "Can not open $input_file: $!";
+  
+
+  while (<FH>) {
+    chomp;
+    my @fields = split("\t");
+    my $var_id = $fields[0];
+    my $var_set_id = $fields[1];
+    my @sets;
+    if (exists $concat_sets{$var_id}) {
+      $concat_sets{$var_id} = [] unless ref $concat_sets{$var_id} eq 'ARRAY';
```

Comment:
If $concat_sets{$var_id} extists, it is reset to an empty array and the original value lost. 

Skipping lines 346,347, 352,353,354 would avoid this